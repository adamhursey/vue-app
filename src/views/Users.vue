<template>
  <div>
    <div class="users" v-for="(user, index) in usersList" :key="index">
      <User v-bind:userId="user.id" v-bind:userName="user.name" />
    </div>
  </div>
</template>

<script>
import axios from "axios";
import User from "../components/User";

export default {
  name: "Users",

  components: {
    User
  },

  data() {
    return {
      usersList: []
    };
  },

  // Fetches all Users from API for user list
  created() {
    console.log("created");
    try {
      axios
        .get("https://jsonplaceholder.typicode.com/users/")
        .then(response => {
          this.usersList = response.data;
        });
    } catch (e) {
      console.log("Could not fetch Users", e);
    }
  }
};
</script>
