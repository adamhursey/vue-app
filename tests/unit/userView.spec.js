import { shallowMount } from "@vue/test-utils";
import Users from "@/views/Users";

describe("UserView", () => {
  it("renders the component", () => {
    // arrange
    const wrapper = shallowMount(Users);

    // assert
    expect(wrapper.html()).toMatchSnapshot();
  });
});
