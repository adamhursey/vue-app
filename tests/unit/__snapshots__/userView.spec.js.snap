// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserView renders the component 1`] = `
<div class="home"><img alt="Vue logo" src="../assets/logo.png">
  <helloworld-stub msg="Welcome to Your Vue.js App"></helloworld-stub>
</div>
`;
